CommonData/dataset.cpp

```diff
@@ -619,3 +619,9 @@ bool DataSet::initColumnWithStrings(int colIndex, const std::string & newName, c
 	return anyChanges || column->type() != prevType;
 }
 
+bool DataSet::isFilterNameFree(const std::string& filterName)
+{
+	return -1 == db().filterGetId(filterName);
```

Comment:
Makes more sense to have this on `Filter` because the name is independent of the DataSet in question.

Reply:
Why did you make it non-static?